### ISSUE TITLE
TensorRT on Multi-GPU platform

### ISSUE THREAD

**Issue by shunfan-shao** [wang-xinyu/tensorrtx#865]
## Env

- Jetson TX 2

## About this repo

- master
- yolov5

## Your problem

I came across [How to Implement Multi-GPU Processing](https://github.com/wang-xinyu/tensorrtx/blob/master/tutorials/multi_GPU_processing.md). I think the question is answered in [TensorRT FAQ](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#faq).

**Q: How do I use TensorRT on multiple GPUs?**
> A: Each ICudaEngine object is bound to a specific GPU when it is instantiated, either by the builder or on deserialization. To select the GPU, use cudaSetDevice() before calling the builder or deserializing the engine. Each IExecutionContext is bound to the same GPU as the engine from which it was created. When calling execute() or enqueue(), ensure that the thread is associated with the correct device by calling cudaSetDevice() if necessary.

Basically, each trt engine is bounded to one GPU. By multi GPU on FAQ, they mean running multiple inference engines on different devices. You could still try to make one inference on multiple GPU work, but I think it is not ideal. 

The tutorial guide in `tensorrtx` is a great guide but does not seem to be very close relevant (eg: we use stream because we can overlap memory/compute operations, we use `cudaMallocHost` because we want to take advantage of page-locked memory). These are very general guide that would be helpful in most cases, but not quite what I am expecting in a multi-GPU guide.
